test_runner/regress/test_tenant_size.py

```diff
@@ -0,0 +1,342 @@
+
+    http_client = env.pageserver.http_client()
+
+    collected_responses: List[Tuple[Lsn, Any]] = []
+
+    with env.postgres.create_start(branch_name, tenant_id=tenant_id) as pg:
+        with pg.cursor() as cur:
+            cur.execute("CREATE TABLE t0 (i BIGINT NOT NULL)")
+
+        batch_size = 100
+
+        i = 0
+        while True:
+            with pg.cursor() as cur:
+                cur.execute(
+                    f"INSERT INTO t0(i) SELECT i FROM generate_series({batch_size} * %s, ({batch_size} * (%s + 1)) - 1) s(i)",
+                    (i, i),
+                )
+
+            i += 1
+
+            current_lsn = wait_for_last_flush_lsn(env, pg, tenant_id, timeline_id)
+
+            size, _ = get_tenant_size(http_client, tenant_id)
+
+            if len(collected_responses) > 0:
+                last_size = collected_responses[-1][1]
+                if size == 0:
+                    assert last_size == 0
+                else:
+                    assert size > last_size
+
+            collected_responses.append((current_lsn, size))
+
+            if len(collected_responses) > 2:
+                break
+
+        while True:
+            with pg.cursor() as cur:
+                cur.execute(
+                    f"UPDATE t0 SET i = -i WHERE i IN (SELECT i FROM t0 WHERE i > 0 LIMIT {batch_size})"
+                )
+                updated = cur.rowcount
+
+            if updated == 0:
+                break
+
+            current_lsn = wait_for_last_flush_lsn(env, pg, tenant_id, timeline_id)
+
+            size, _ = get_tenant_size(http_client, tenant_id)
+            assert size > collected_responses[-1][1]
+            collected_responses.append((current_lsn, size))
+
+        while True:
+            with pg.cursor() as cur:
+                cur.execute(f"DELETE FROM t0 WHERE i IN (SELECT i FROM t0 LIMIT {batch_size})")
+                deleted = cur.rowcount
+
+            if deleted == 0:
+                break
+
+            current_lsn = wait_for_last_flush_lsn(env, pg, tenant_id, timeline_id)
+
+            size, _ = get_tenant_size(http_client, tenant_id)
+            # even though rows have been deleted, the size should not decrease
+            assert size > collected_responses[-1][1]
+            collected_responses.append((current_lsn, size))
+
+        with pg.cursor() as cur:
+            cur.execute("DROP TABLE t0")
+
+        current_lsn = wait_for_last_flush_lsn(env, pg, tenant_id, timeline_id)
+
+        size, _ = get_tenant_size(http_client, tenant_id)
+
+        assert size > collected_responses[-1][1]
+        collected_responses.append((current_lsn, size))
+
+    for lsn, size in collected_responses:
+        log.info(f"collected: {lsn}, {size}")
+
+    env.pageserver.stop()
+    env.pageserver.start()
+
+    size_after = get_tenant_size(http_client, tenant_id)[0]
+
+    assert (
+        size_after == collected_responses[-1][1]
```

Comment:
NIT: feels more readable to extract `collected_responses[-1][1]` into a separate variable with the name.

Reply:
Did this, and will push next the `last_size` => `prev` as well.